### DISCUSSION TITLE
Read access violation

### DISCUSSION THREAD

**Discussion by GabrielSenekovic** [mikke89/RmlUi#499] [Q&A]
Hello! I am having trouble implementing rmlUI into my project. I have followed the tutorial and the examples. First, the default file interface did not seem to want to work. The game crashed when it tried to Render() a context. The program enters LoadTexture(), where every single parameter has "read access violation" on it's values. I understand that LoadTexture() is supposed to load files when required, but there is no way to tell what files that it is trying to load. I am not personally trying to load any images. I am simply calling context.Render(). It has an empty document in it. The error persists when the document is removed. I have tried removing and adding things for close to 5 hours now, and nothing fixes the issue. I can provide code, but I am not sure what it would solve. I don't really know where to go next from here.
I appreciate any response. Thank you.

**Comment by mikke89** [comment]
Hello, and welcome! Sorry to hear about the trouble, sounds quite frustrating.

It's hard to know exactly what's going on based ok this information alone. However, it does sound like something that can happen with memory corruption. Can you try to run it under adress sanitizer? That will help diagnose many such issues.

Another common cause is lifetime issues, resulting in use-after-free. Please ensure that everything runs in correct order. Both while setting up the interfaces, and while initializing the library. See specifically this document: https://mikke89.github.io/RmlUiDoc/pages/cpp_manual/main_loop.html

Another possible issue could be binary incompatibility. Ensure that the library you are linking to matches your setup in terms of compiler and compiler flags, plus compile-time definitions for the library.
